



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.0.4">
    
    
      
        <title>RNA-Seq - Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application-709eec9410.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      <script src="../assets/javascripts/modernizr-e826f8942a.js"></script>
    
    
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    
    
    
      <body data-md-color-primary="pink" data-md-color-accent="pink">
    
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
            
            RNA-Seq
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Bioinformatics Tutorials
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../wms/" title="Whole Metagenome Sequencing" class="md-nav__link">
      Whole Metagenome Sequencing
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        RNA-Seq
      </label>
    
    <a href="./" title="RNA-Seq" class="md-nav__link md-nav__link--active">
      RNA-Seq
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#load-salmon" title="Load salmon" class="md-nav__link">
    Load salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indexing-transcriptome" title="Indexing transcriptome" class="md-nav__link">
    Indexing transcriptome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quantify-reads-using-salmon" title="Quantify reads using salmon" class="md-nav__link">
    Quantify reads using salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-read-counts-using-tximport" title="Import read counts using tximport" class="md-nav__link">
    Import read counts using tximport
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-using-deseq2" title="Differential expression using DESeq2" class="md-nav__link">
    Differential expression using DESeq2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#kegg-pathway-analysis" title="KEGG pathway analysis" class="md-nav__link">
    KEGG pathway analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#thanks" title="Thanks" class="md-nav__link">
    Thanks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#load-salmon" title="Load salmon" class="md-nav__link">
    Load salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indexing-transcriptome" title="Indexing transcriptome" class="md-nav__link">
    Indexing transcriptome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quantify-reads-using-salmon" title="Quantify reads using salmon" class="md-nav__link">
    Quantify reads using salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-read-counts-using-tximport" title="Import read counts using tximport" class="md-nav__link">
    Import read counts using tximport
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-using-deseq2" title="Differential expression using DESeq2" class="md-nav__link">
    Differential expression using DESeq2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#kegg-pathway-analysis" title="KEGG pathway analysis" class="md-nav__link">
    KEGG pathway analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#thanks" title="Thanks" class="md-nav__link">
    Thanks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/rna.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="rna-seq">RNA-Seq<a class="headerlink" href="#rna-seq" title="Permanent link">&para;</a></h1>
<h2 id="load-salmon">Load salmon<a class="headerlink" href="#load-salmon" title="Permanent link">&para;</a></h2>
<div class="codehilite"><pre><span></span><span class="k">module</span> <span class="n">load</span> <span class="n">Salmon</span>
</pre></div>


<h2 id="downloading-the-data">Downloading the data<a class="headerlink" href="#downloading-the-data" title="Permanent link">&para;</a></h2>
<p>For this tutorial we will use the test data from <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004393">this</a> paper:</p>
<blockquote>
<p>Malachi Griffith<em>, Jason R. Walker, Nicholas C. Spies, Benjamin J. Ainscough, Obi L. Griffith</em>. 2015. Informatics for RNA-seq: A web resource for analysis on the cloud. PLoS Comp Biol. 11(8):e1004393.</p>
</blockquote>
<p>The test data consists of two commercially available RNA samples: Universal Human Reference (UHR) and Human Brain Reference (HBR). The UHR is total RNA isolated from a diverse set of 10 cancer cell lines. The HBR is total RNA isolated from the brains of 23 Caucasians, male and female, of varying age but mostly 60-80 years old.</p>
<p>In addition, a spike-in control was used. Specifically we added an aliquot of the ERCC ExFold RNA Spike-In Control Mixes to each sample. The spike-in consists of 92 transcripts that are present in known concentrations across a wide abundance range (from very few copies to many copies). This range allows us to test the degree to which the RNA-seq assay (including all laboratory and analysis steps) accurately reflects the relative abundance of transcript species within a sample. There are two 'mixes' of these transcripts to allow an assessment of differential expression output between samples if you put one mix in each of your two comparisons. In our case, Mix1 was added to the UHR sample, and Mix2 was added to the HBR sample. We also have 3 complete experimental replicates for each sample. This allows us to assess the technical variability of our overall process of producing RNA-seq data in the lab.</p>
<p>For all libraries we prepared low-throughput (Set A) TruSeq Stranded Total RNA Sample Prep Kit libraries with Ribo-Zero Gold to remove both cytoplasmic and mitochondrial rRNA. Triplicate, indexed libraries were made starting with 100ng Agilent/Strategene Universal Human Reference total RNA and 100ng Ambion Human Brain Reference total RNA. The Universal Human Reference replicates received 2 ul of 1:1000 ERCC Mix 1. The Human Brain Reference replicates received 1:1000 ERCC Mix 2. The libraries were quantified with KAPA Library Quantification qPCR and adjusted to the appropriate concentration for sequencing. The triplicate, indexed libraries were then pooled prior to sequencing. Each pool of three replicate libraries were sequenced across 2 lanes of a HiSeq 2000 using paired-end sequence chemistry with 100bp read lengths.</p>
<p>So to summarize we have:</p>
<ul>
<li>UHR + ERCC Spike-In Mix1, Replicate 1</li>
<li>UHR + ERCC Spike-In Mix1, Replicate 2</li>
<li>UHR + ERCC Spike-In Mix1, Replicate 3</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 1</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 2</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 3</li>
</ul>
<p>You can download the data from <a href="http://139.162.178.46/files/tutorials/toy_rna.tar.gz">here</a>.</p>
<p>Unpack the data and go into the toy_rna directory</p>
<div class="codehilite"><pre><span></span>tar xzf toy_rna.tar.gz
cd toy_rna
</pre></div>


<h2 id="indexing-transcriptome">Indexing transcriptome<a class="headerlink" href="#indexing-transcriptome" title="Permanent link">&para;</a></h2>
<div class="codehilite"><pre><span></span>salmon index -t chr22_transcripts.fa -i chr22_index
</pre></div>


<h2 id="quantify-reads-using-salmon">Quantify reads using salmon<a class="headerlink" href="#quantify-reads-using-salmon" title="Permanent link">&para;</a></h2>
<div class="codehilite"><pre><span></span><span class="k">for</span> i in *_R1.fastq.gz
<span class="k">do</span>
   <span class="nv">prefix</span><span class="o">=</span><span class="k">$(</span>basename <span class="nv">$i</span> _R1.fastq.gz<span class="k">)</span>
   salmon quant -i chr22_index --libType A <span class="se">\</span>
          -1 <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R1.fastq.gz -2 <span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>_R2.fastq.gz -o quant/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span><span class="p">;</span>
<span class="k">done</span>
</pre></div>


<p>This loop simply goes through each sample and invokes salmon using fairly basic options:</p>
<ul>
<li>The -i argument tells salmon where to find the index</li>
<li>--libType A tells salmon that it should automatically determine the library type of the sequencing reads (e.g. stranded vs. unstranded etc.)</li>
<li>The -1 and -2 arguments tell salmon where to find the left and right reads for this sample (notice, salmon will accept gzipped FASTQ files directly).</li>
<li>the -o argument specifies the directory where salmon’s quantification results sould be written.</li>
</ul>
<p>Salmon exposes many different options to the user that enable extra features or modify default behavior. However, the purpose and behavior of all of those options is beyond the scope of this introductory tutorial. You can read about salmon’s many options in the <a href="http://salmon.readthedocs.io/en/latest/">documentation</a>.</p>
<p>After the salmon commands finish running, you should have a directory named <code>quant</code>, which will have a sub-directory for each sample. These sub-directories contain the quantification results of salmon, as well as a lot of other information salmon records about the sample and the run. The main output file (called quant.sf) is rather self-explanatory. For example, take a peek at the quantification file for sample <code>HBR_Rep1</code> in <code>quant/HBR_Rep1/quant.sf</code> and you’ll see a simple TSV format file listing the name (Name) of each transcript, its length (Length), effective length (EffectiveLength) (more details on this in the documentation), and its abundance in terms of Transcripts Per Million (TPM) and estimated number of reads (NumReads) originating from this transcript.</p>
<h2 id="import-read-counts-using-tximport">Import read counts using tximport<a class="headerlink" href="#import-read-counts-using-tximport" title="Permanent link">&para;</a></h2>
<p>Using the tximport R package, you can import salmon’s transcript-level quantifications and optionally aggregate them to the gene level for gene-level differential expression analysis.</p>
<p>First, open up your favourite R IDE and install the necessary packages:</p>
<div class="codehilite"><pre><span></span><span class="kn">source</span><span class="p">(</span><span class="s">&quot;https://bioconductor.org/biocLite.R&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;tximport&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;GenomicFeatures&quot;</span><span class="p">)</span>

install.packages<span class="p">(</span><span class="s">&quot;readr&quot;</span><span class="p">)</span>
</pre></div>


<p>Then load the modules:</p>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>tximport<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>GenomicFeatures<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>readr<span class="p">)</span>
</pre></div>


<p>Salmon did the quantifiation of the transcript level. We want to see which genes are differentially expressed, so we need to link the transcript names to the gene names. We can use our .gtf annotation for that, and the GenomicFeatures package:</p>
<div class="codehilite"><pre><span></span>txdb <span class="o">&lt;-</span> makeTxDbFromGFF<span class="p">(</span><span class="s">&quot;chr22_genes.gtf&quot;</span><span class="p">)</span>
k <span class="o">&lt;-</span> keys<span class="p">(</span>txdb<span class="p">,</span> keytype <span class="o">=</span> <span class="s">&quot;GENEID&quot;</span><span class="p">)</span>
tx2gene <span class="o">&lt;-</span> select<span class="p">(</span>txdb<span class="p">,</span> keys <span class="o">=</span> k<span class="p">,</span> keytype <span class="o">=</span> <span class="s">&quot;GENEID&quot;</span><span class="p">,</span> columns <span class="o">=</span> <span class="s">&quot;TXNAME&quot;</span><span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>tx2gene<span class="p">)</span>
</pre></div>


<p>Now we can import the salmon quantification. First, download the file with sample descriptions from <a href="https://raw.githubusercontent.com/HadrienG/tutorials/master/data/samples.txt">here</a> and put it in the toy_rna directory. Then, use that file to load the corresponding quantification data.</p>
<div class="codehilite"><pre><span></span>samples <span class="o">&lt;-</span> read.table<span class="p">(</span><span class="s">&quot;samples.txt&quot;</span><span class="p">,</span> header <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
files <span class="o">&lt;-</span> <span class="kp">file.path</span><span class="p">(</span><span class="s">&quot;quant&quot;</span><span class="p">,</span> samples<span class="o">$</span><span class="kp">sample</span><span class="p">,</span> <span class="s">&quot;quant.sf&quot;</span><span class="p">)</span>
<span class="kp">names</span><span class="p">(</span>files<span class="p">)</span> <span class="o">&lt;-</span> <span class="kp">paste0</span><span class="p">(</span>samples<span class="o">$</span><span class="kp">sample</span><span class="p">)</span>
txi.salmon <span class="o">&lt;-</span> tximport<span class="p">(</span>files<span class="p">,</span> type <span class="o">=</span> <span class="s">&quot;salmon&quot;</span><span class="p">,</span> tx2gene <span class="o">=</span> tx2gene<span class="p">,</span> reader <span class="o">=</span> read_tsv<span class="p">)</span>
</pre></div>


<p>Take a look at the data:</p>
<div class="codehilite"><pre><span></span><span class="kp">head</span><span class="p">(</span>txi.salmon<span class="o">$</span>counts<span class="p">)</span>
</pre></div>


<h2 id="differential-expression-using-deseq2">Differential expression using DESeq2<a class="headerlink" href="#differential-expression-using-deseq2" title="Permanent link">&para;</a></h2>
<p>Install the necessary package:</p>
<div class="codehilite"><pre><span></span>biocLite<span class="p">(</span><span class="s">&#39;DESeq2&#39;</span><span class="p">)</span>
</pre></div>


<p>Then load it:</p>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>DESeq2<span class="p">)</span>
</pre></div>


<p>Instantiate the DESeqDataSet and generate result table. See <code>?DESeqDataSetFromTximport</code> and <code>?DESeq</code> for more information about the steps performed by the program.</p>
<div class="codehilite"><pre><span></span>dds <span class="o">&lt;-</span> DESeqDataSetFromTximport<span class="p">(</span>txi.salmon<span class="p">,</span> samples<span class="p">,</span> <span class="o">~</span>condition<span class="p">)</span>
dds <span class="o">&lt;-</span> DESeq<span class="p">(</span>dds<span class="p">)</span>
res <span class="o">&lt;-</span> results<span class="p">(</span>dds<span class="p">)</span>
</pre></div>


<p>Run the <code>summary</code> command to get an idea of how many genes are up- and downregulated between the two conditions:</p>
<p><code>summary(res)</code></p>
<p>DESeq uses a negative binomial distribution. Such distributions have two parameters: mean and dispersion. The dispersion is a parameter describing how much the variance deviates from the mean.</p>
<p>You can read more about the methods used by DESeq2 in the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">paper</a> or the <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq/inst/doc/DESeq.pdf">vignette</a></p>
<p>Plot dispersions:</p>
<div class="codehilite"><pre><span></span>plotDispEsts<span class="p">(</span>dds<span class="p">,</span> main<span class="o">=</span><span class="s">&quot;Dispersion plot&quot;</span><span class="p">)</span>
</pre></div>


<p>For clustering and heatmaps, we need to log transform our data:</p>
<div class="codehilite"><pre><span></span>rld <span class="o">&lt;-</span> rlogTransformation<span class="p">(</span>dds<span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>assay<span class="p">(</span>rld<span class="p">))</span>
</pre></div>


<p>Then, we create a sample distance heatmap:</p>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>RColorBrewer<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>gplots<span class="p">)</span> <span class="c1"># you may need to install this package</span>

<span class="p">(</span>mycols <span class="o">&lt;-</span> brewer.pal<span class="p">(</span><span class="m">8</span><span class="p">,</span> <span class="s">&quot;Dark2&quot;</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="kp">length</span><span class="p">(</span><span class="kp">unique</span><span class="p">(</span>samples<span class="o">$</span>condition<span class="p">))])</span>
sampleDists <span class="o">&lt;-</span> <span class="kp">as.matrix</span><span class="p">(</span>dist<span class="p">(</span><span class="kp">t</span><span class="p">(</span>assay<span class="p">(</span>rld<span class="p">))))</span>
heatmap.2<span class="p">(</span><span class="kp">as.matrix</span><span class="p">(</span>sampleDists<span class="p">),</span> key<span class="o">=</span><span class="bp">F</span><span class="p">,</span> trace<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">,</span>
          col<span class="o">=</span>colorpanel<span class="p">(</span><span class="m">100</span><span class="p">,</span> <span class="s">&quot;black&quot;</span><span class="p">,</span> <span class="s">&quot;white&quot;</span><span class="p">),</span>
          ColSideColors<span class="o">=</span>mycols<span class="p">[</span>samples<span class="o">$</span>condition<span class="p">],</span>
          RowSideColors<span class="o">=</span>mycols<span class="p">[</span>samples<span class="o">$</span>condition<span class="p">],</span>
          margin<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">10</span><span class="p">,</span> <span class="m">10</span><span class="p">),</span> main<span class="o">=</span><span class="s">&quot;Sample Distance Matrix&quot;</span><span class="p">)</span>
</pre></div>


<p>We can also plot a PCA:</p>
<div class="codehilite"><pre><span></span>DESeq2<span class="o">::</span>plotPCA<span class="p">(</span>rld<span class="p">,</span> intgroup<span class="o">=</span><span class="s">&quot;condition&quot;</span><span class="p">)</span>
</pre></div>


<p>It is time to look at some p-values:</p>
<div class="codehilite"><pre><span></span><span class="kp">table</span><span class="p">(</span>res<span class="o">$</span>padj<span class="o">&lt;</span><span class="m">0.05</span><span class="p">)</span>
res <span class="o">&lt;-</span> res<span class="p">[</span><span class="kp">order</span><span class="p">(</span>res<span class="o">$</span>padj<span class="p">),</span> <span class="p">]</span>
resdata <span class="o">&lt;-</span> <span class="kp">merge</span><span class="p">(</span><span class="kp">as.data.frame</span><span class="p">(</span>res<span class="p">),</span> <span class="kp">as.data.frame</span><span class="p">(</span>counts<span class="p">(</span>dds<span class="p">,</span> normalized<span class="o">=</span><span class="kc">TRUE</span><span class="p">)),</span> by<span class="o">=</span><span class="s">&quot;row.names&quot;</span><span class="p">,</span> sort<span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>
<span class="kp">names</span><span class="p">(</span>resdata<span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s">&quot;Gene&quot;</span>
<span class="kp">head</span><span class="p">(</span>resdata<span class="p">)</span>
</pre></div>


<p>Examine plot of p-values, the MA plot and the Volcano Plot:</p>
<div class="codehilite"><pre><span></span>hist<span class="p">(</span>res<span class="o">$</span>pvalue<span class="p">,</span> breaks<span class="o">=</span><span class="m">50</span><span class="p">,</span> col<span class="o">=</span><span class="s">&quot;grey&quot;</span><span class="p">)</span>
DESeq2<span class="o">::</span>plotMA<span class="p">(</span>dds<span class="p">,</span> ylim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-1</span><span class="p">,</span><span class="m">1</span><span class="p">),</span> cex<span class="o">=</span><span class="m">1</span><span class="p">)</span>

<span class="c1"># Volcano plot</span>
<span class="kp">with</span><span class="p">(</span>res<span class="p">,</span> plot<span class="p">(</span>log2FoldChange<span class="p">,</span> <span class="o">-</span><span class="kp">log10</span><span class="p">(</span>pvalue<span class="p">),</span> pch<span class="o">=</span><span class="m">20</span><span class="p">,</span> main<span class="o">=</span><span class="s">&quot;Volcano plot&quot;</span><span class="p">,</span> xlim<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">-2.5</span><span class="p">,</span><span class="m">2</span><span class="p">)))</span>
<span class="kp">with</span><span class="p">(</span><span class="kp">subset</span><span class="p">(</span>res<span class="p">,</span> padj<span class="o">&lt;</span><span class="m">.05</span> <span class="p">),</span> points<span class="p">(</span>log2FoldChange<span class="p">,</span> <span class="o">-</span><span class="kp">log10</span><span class="p">(</span>pvalue<span class="p">),</span> pch<span class="o">=</span><span class="m">20</span><span class="p">,</span> col<span class="o">=</span><span class="s">&quot;red&quot;</span><span class="p">))</span>
</pre></div>


<h2 id="kegg-pathway-analysis">KEGG pathway analysis<a class="headerlink" href="#kegg-pathway-analysis" title="Permanent link">&para;</a></h2>
<p>As always, install and load the necessary packages:</p>
<div class="codehilite"><pre><span></span>biocLite<span class="p">(</span><span class="s">&quot;AnnotationDbi&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;org.Hs.eg.db&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;pathview&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;gage&quot;</span><span class="p">)</span>
biocLite<span class="p">(</span><span class="s">&quot;gageData&quot;</span><span class="p">)</span>

<span class="kn">library</span><span class="p">(</span>AnnotationDbi<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>org.Hs.eg.db<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>pathview<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>gage<span class="p">)</span>
<span class="kn">library</span><span class="p">(</span>gageData<span class="p">)</span>
</pre></div>


<p>Let’s use the <code>mapIds</code> function to add more columns to the results. The row.names of our results table has the Ensembl gene ID (our key), so we need to specify  <code>keytype=ENSEMBL</code>. The column argument tells the <code>mapIds</code> function which information we want, and the <code>multiVals</code> argument tells the function what to do if there are multiple possible values for a single input value. Here we ask to just give us back the first one that occurs in the database. Let’s get the Entrez IDs, gene symbols, and full gene names.</p>
<div class="codehilite"><pre><span></span>res<span class="o">$</span>symbol <span class="o">&lt;-</span> mapIds<span class="p">(</span>org.Hs.eg.db<span class="p">,</span>
                     keys<span class="o">=</span><span class="kp">row.names</span><span class="p">(</span>res<span class="p">),</span>
                     column<span class="o">=</span><span class="s">&quot;SYMBOL&quot;</span><span class="p">,</span>
                     keytype<span class="o">=</span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
                     multiVals<span class="o">=</span><span class="s">&quot;first&quot;</span><span class="p">)</span>
res<span class="o">$</span>entrez <span class="o">&lt;-</span> mapIds<span class="p">(</span>org.Hs.eg.db<span class="p">,</span>
                     keys<span class="o">=</span><span class="kp">row.names</span><span class="p">(</span>res<span class="p">),</span>
                     column<span class="o">=</span><span class="s">&quot;ENTREZID&quot;</span><span class="p">,</span>
                     keytype<span class="o">=</span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
                     multiVals<span class="o">=</span><span class="s">&quot;first&quot;</span><span class="p">)</span>
res<span class="o">$</span>name <span class="o">&lt;-</span> mapIds<span class="p">(</span>org.Hs.eg.db<span class="p">,</span>
                     keys<span class="o">=</span><span class="kp">row.names</span><span class="p">(</span>res<span class="p">),</span>
                     column<span class="o">=</span><span class="s">&quot;GENENAME&quot;</span><span class="p">,</span>
                     keytype<span class="o">=</span><span class="s">&quot;ENSEMBL&quot;</span><span class="p">,</span>
                     multiVals<span class="o">=</span><span class="s">&quot;first&quot;</span><span class="p">)</span>

<span class="kp">head</span><span class="p">(</span>res<span class="p">)</span>
</pre></div>


<p>We’re going to use the <a href="https://bioconductor.org/packages/release/bioc/html/gage.html">gage</a> package for pathway analysis, and the <a href="https://bioconductor.org/packages/release/bioc/html/pathview.html">pathview</a> package to draw a pathway diagram.</p>
<p>The gageData package has pre-compiled databases mapping genes to KEGG pathways and GO terms for common organisms:</p>
<div class="codehilite"><pre><span></span>data<span class="p">(</span>kegg.sets.hs<span class="p">)</span>
data<span class="p">(</span>sigmet.idx.hs<span class="p">)</span>
kegg.sets.hs <span class="o">&lt;-</span> kegg.sets.hs<span class="p">[</span>sigmet.idx.hs<span class="p">]</span>
<span class="kp">head</span><span class="p">(</span>kegg.sets.hs<span class="p">,</span> <span class="m">3</span><span class="p">)</span>
</pre></div>


<p>Run the pathway analysis. See help on the gage function with <code>?gage</code>. Specifically, you might want to try changing the value of same.dir.</p>
<div class="codehilite"><pre><span></span>foldchanges <span class="o">&lt;-</span> res<span class="o">$</span>log2FoldChange
<span class="kp">names</span><span class="p">(</span>foldchanges<span class="p">)</span> <span class="o">&lt;-</span> res<span class="o">$</span>entrez
keggres <span class="o">&lt;-</span> gage<span class="p">(</span>foldchanges<span class="p">,</span> gsets<span class="o">=</span>kegg.sets.hs<span class="p">,</span> same.dir<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="kp">lapply</span><span class="p">(</span>keggres<span class="p">,</span> <span class="kp">head</span><span class="p">)</span>
</pre></div>


<p>Pull out the top 5 upregulated pathways, then further process that just to get the IDs. We’ll use these KEGG pathway IDs downstream for plotting. The <code>dplyr</code> package is required to use the pipe (<code>%&gt;%</code>) construct.</p>
<div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span>dplyr<span class="p">)</span>

<span class="c1"># Get the pathways</span>
keggrespathways <span class="o">&lt;-</span> <span class="kt">data.frame</span><span class="p">(</span>id<span class="o">=</span><span class="kp">rownames</span><span class="p">(</span>keggres<span class="o">$</span>greater<span class="p">),</span> keggres<span class="o">$</span>greater<span class="p">)</span> <span class="o">%&gt;%</span>
  tbl_df<span class="p">()</span> <span class="o">%&gt;%</span>
  filter<span class="p">(</span>row_number<span class="p">()</span><span class="o">&lt;=</span><span class="m">5</span><span class="p">)</span> <span class="o">%&gt;%</span>
  <span class="m">.</span><span class="o">$</span>id <span class="o">%&gt;%</span>
  <span class="kp">as.character</span><span class="p">()</span>
keggrespathways

<span class="c1"># Get the IDs.</span>
keggresids <span class="o">&lt;-</span> <span class="kp">substr</span><span class="p">(</span>keggrespathways<span class="p">,</span> start<span class="o">=</span><span class="m">1</span><span class="p">,</span> stop<span class="o">=</span><span class="m">8</span><span class="p">)</span>
keggresids
</pre></div>


<p>Finally, the <code>pathview()</code> function in the pathview package makes the plots. Let’s write a function so we can loop through and draw plots for the top 5 pathways we created above.</p>
<div class="codehilite"><pre><span></span><span class="c1"># Define plotting function for applying later</span>
plot_pathway <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>pid<span class="p">)</span> pathview<span class="p">(</span>gene.data<span class="o">=</span>foldchanges<span class="p">,</span> pathway.id<span class="o">=</span>pid<span class="p">,</span> species<span class="o">=</span><span class="s">&quot;hsa&quot;</span><span class="p">,</span> new.signature<span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>

<span class="c1"># Unload dplyr since it conflicts with the next line</span>
<span class="kn">detach</span><span class="p">(</span><span class="s">&quot;package:dplyr&quot;</span><span class="p">,</span> unload<span class="o">=</span><span class="bp">T</span><span class="p">)</span>

<span class="c1"># plot multiple pathways (plots saved to disk and returns a throwaway list object)</span>
tmp <span class="o">&lt;-</span> <span class="kp">sapply</span><span class="p">(</span>keggresids<span class="p">,</span> <span class="kr">function</span><span class="p">(</span>pid<span class="p">)</span> pathview<span class="p">(</span>gene.data<span class="o">=</span>foldchanges<span class="p">,</span> pathway.id<span class="o">=</span>pid<span class="p">,</span> species<span class="o">=</span><span class="s">&quot;hsa&quot;</span><span class="p">))</span>
</pre></div>


<h4 id="thanks">Thanks<a class="headerlink" href="#thanks" title="Permanent link">&para;</a></h4>
<p>This material was inspired by Stephen Turner's blog post:</p>
<blockquote>
<p>Tutorial: RNA-seq differential expression &amp; pathway analysis with Sailfish, DESeq2, GAGE, and Pathview: http://www.gettinggeneticsdone.com/2015/12/tutorial-rna-seq-differential.html</p>
</blockquote>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../meta_assembly/" title="Metagenome assembly" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metagenome assembly
              </span>
            </div>
          </a>
        
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application-8e4952e681.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:".."}})</script>
      
    
    
      
    
  </body>
</html>