



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.0.4">
    
    
      
        <title>Mapping and Variant Calling - Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application-709eec9410.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      <script src="../assets/javascripts/modernizr-e826f8942a.js"></script>
    
    
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    
    
    
      <body data-md-color-primary="pink" data-md-color-accent="pink">
    
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
            
            Mapping and Variant Calling
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Bioinformatics Tutorials
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Mapping and Variant Calling
      </label>
    
    <a href="./" title="Mapping and Variant Calling" class="md-nav__link md-nav__link--active">
      Mapping and Variant Calling
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#read-mapping" title="Read mapping" class="md-nav__link">
    Read mapping
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#downloading-a-reference" title="downloading a reference" class="md-nav__link">
    downloading a reference
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#indexing-the-reference" title="indexing the reference" class="md-nav__link">
    indexing the reference
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#aligning-reads" title="aligning reads" class="md-nav__link">
    aligning reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#visualising-the-alignment" title="visualising the alignment" class="md-nav__link">
    visualising the alignment
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling" title="Variant Calling" class="md-nav__link">
    Variant Calling
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sorting-and-indexing-a-bam-file" title="Sorting and indexing a bam file" class="md-nav__link">
    Sorting and indexing a bam file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../wms/" title="Whole Metagenome Sequencing" class="md-nav__link">
      Whole Metagenome Sequencing
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#read-mapping" title="Read mapping" class="md-nav__link">
    Read mapping
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#downloading-a-reference" title="downloading a reference" class="md-nav__link">
    downloading a reference
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#indexing-the-reference" title="indexing the reference" class="md-nav__link">
    indexing the reference
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#aligning-reads" title="aligning reads" class="md-nav__link">
    aligning reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#visualising-the-alignment" title="visualising the alignment" class="md-nav__link">
    visualising the alignment
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling" title="Variant Calling" class="md-nav__link">
    Variant Calling
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sorting-and-indexing-a-bam-file" title="Sorting and indexing a bam file" class="md-nav__link">
    Sorting and indexing a bam file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/mapping.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="mapping-and-variant-calling">Mapping and Variant Calling<a class="headerlink" href="#mapping-and-variant-calling" title="Permanent link">&para;</a></h1>
<p>In this practical you will learn to map NGS reads to a reference sequence, check the output using a viewer software and investigate some aspects of the results. You will be using the read data from the <a href="../qc/">Quality Control</a> practical.</p>
<p>EHEC O157 strains generally carry a large virulence plasmid, pO157. Plasmids are circular genetic elements that many bacteria carry in addition to their chromosomes. This particular plasmid encodes a number of proteins which are known or suspected to be involved in the ability to cause severe disease in infected humans. Your task in this practical is to map your prepared read set to a reference sequence of the virulence plasmid, to determine if the pO157 plasmid is present in the St. Louis outbreak strain.</p>
<h2 id="read-mapping">Read mapping<a class="headerlink" href="#read-mapping" title="Permanent link">&para;</a></h2>
<h3 id="downloading-a-reference">downloading a reference<a class="headerlink" href="#downloading-a-reference" title="Permanent link">&para;</a></h3>
<p>You will need a reference sequence to map your reads to. You can download the reference from <a href="../data/pO157_Sakai.fasta">here</a></p>
<p>or from the command-line with</p>
<p><code>wget https://raw.githubusercontent.com/HadrienG/tutorials/master/data/pO157_Sakai.fasta</code></p>
<p>This file contains the sequence of the pO157 plasmid from the Sakai outbreak strain of E. coli O157. In contrast to the strain we are working on, this strain is available as a finished genome, i.e. the whole sequence of both the single chromosome and the large virulence plasmid are known.</p>
<h3 id="indexing-the-reference">indexing the reference<a class="headerlink" href="#indexing-the-reference" title="Permanent link">&para;</a></h3>
<p>Before aligning the reads against a reference, it is necessary to build an index of that reference:</p>
<div class="codehilite"><pre><span></span><span class="k">module</span> <span class="n">load</span> <span class="n">bowtie2</span>
<span class="n">bowtie2-build</span> <span class="n">pO157_Sakai</span>.<span class="n">fasta</span> <span class="n">pO157_Sakai</span>
</pre></div>


<h3 id="aligning-reads">aligning reads<a class="headerlink" href="#aligning-reads" title="Permanent link">&para;</a></h3>
<p><code>bowtie2 -x pO157_Sakai -1 reads_1.fastq -2 reads_2.fastq -S output.sam</code></p>
<p>The output of the mapping will be in SAM format. you can find a brief explanation of the SAM format <a href="files_formats.md">here</a></p>
<h3 id="visualising-the-alignment">visualising the alignment<a class="headerlink" href="#visualising-the-alignment" title="Permanent link">&para;</a></h3>
<p>To view the outcome of the read mapping, we will use a program called Tablet, that can be run without administrated privileges. Download it <a href="https://ics.hutton.ac.uk/tablet/">here</a>.</p>
<p>Start the program and select Red Button – Open. Choose your SAM-file and pO157_Sakai.fasta as a reference.</p>
<p><img alt="tablet1" src="../images/tablet1.png" /></p>
<p>Select the only contig to the left.</p>
<p><img alt="tablet2" src="../images/tablet2.png" /></p>
<p>Next, download pO157_Sakai.gff from <a href="../data/pO157_Sakai.gff">here</a>. Select import features in Tablet and import pO157_Sakai.gff. This file contains annotations, i.e. what is encoded in each part of the DNA sequence. Two tracks (CDS + GENE) will be added.</p>
<p>Navigate the mapping using the zoom and pan left/right etc. Under Colour schemes you can highlight bases that do not match the reference. Holding your pointer over a CDS will show you a description of the genetic region.</p>
<p><img alt="tablet3" src="../images/tablet3.png" /></p>
<p>Does the mapping data confirm the presence of an intact pO157 plasmid in the St. Louis outbreak strain?</p>
<p>You will note that one region has extremely high coverage. This region corresponds to two adjacent CDSs</p>
<p>Check their identity, what do they have in common?</p>
<p>Extract the sequence of the biggest of these two by right-clicking on it and selecting copy reference subsequence to clipboard</p>
<p>Your read data was from the whole genome, but you are only mapping to a reference for the pO157 plasmid. Go to BLAST and use nucleotide BLAST to check if the problematic region is present on the chromosome of the reference strain Sakai (BA000007.2) using the Align two sequences option with your subsequence copied from Tablet and BA000007.2. You don’t have to enter the whole genome sequence, just the code.</p>
<p><img alt="blast1" src="../images/blast1.png" /></p>
<p>How many matches do you find on the chromosome?</p>
<p>Why do you think the coverage is so high for this region in the mapping to the pO157 reference?</p>
<h2 id="variant-calling">Variant Calling<a class="headerlink" href="#variant-calling" title="Permanent link">&para;</a></h2>
<p>A frequent application for mapping reads is variant calling, i.e. finding positions where the reference and your sequences differ. Single nucleotide polymorphism (SNP)-based typing is particularly popular and used for a broad range of applications. For an EHEC O157 outbreak you could use it to identify the source, for instance. Indels are insertions and deletions in the mapped data compared to the reference.</p>
<p>You could find SNPs by just staring at the alignment in Tablet. A variant caller is a program that automates this process, finding variable positions anywhere in the sequence and weighing the evidence for a true variant (number of reads agreeing, quality of those reads etc.) We will be using SAMtools for variant calling.</p>
<h3 id="sorting-and-indexing-a-bam-file">Sorting and indexing a bam file<a class="headerlink" href="#sorting-and-indexing-a-bam-file" title="Permanent link">&para;</a></h3>
<p>First you need to sort the reads in the BAM-file so the variant caller can easily find the relevant ones for a given position in the reference sequence.</p>
<p>You may need to convert your sam file into a bam file first:</p>
<p><code>samtools view -bS -o $output.bam $input.sam</code></p>
<p>then sort and index the bam file:</p>
<div class="codehilite"><pre><span></span>samtools sort $output.bam &gt; $output_sorted.bam
samtools index $output_sorted.bam
</pre></div>


<p>Now we can perform the SNP and Indel calling:</p>
<p><code>samtools mpileup -uv -o output.vcf -f pO157_Sakai.fasta test_sorted.bam</code></p>
<p>You can read about the structure of vcf files here</p>
<p>How many SNPs did the variant caller find? Did you find any indels?</p>
<p>Each variant in the vcf comes with a lot of different quality metrics from the variant calling process. The “QUAL” column tells you how confident the variant caller is feeling about that variant.</p>
<p>Use the coordinates from the vcf-file and manually check a high quality SNP and a low quality SNP in your Tablet view of the sam file. (Hint: there is a “Jump to base” button that can make it easier for you.)</p>
<p>If you have time:
Read the original paper about the St. Louis outbreak investigation, in particular check the supplemental material and methods to see how the authors performed read mapping and variant calling.</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../qc/" title="Quality Control and Trimming" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Quality Control and Trimming
              </span>
            </div>
          </a>
        
        
          <a href="../assembly/" title="De-novo Genome Assembly" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                De-novo Genome Assembly
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application-8e4952e681.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:".."}})</script>
      
    
    
      
    
  </body>
</html>