



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.0.4">
    
    
      
        <title>RNA-Seq analysis - SGBC Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application-709eec9410.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      <script src="../assets/javascripts/modernizr-e826f8942a.js"></script>
    
    
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    
    
    
      <body data-md-color-primary="pink" data-md-color-accent="pink">
    
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="SGBC Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
            
            RNA-Seq analysis
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    SGBC Bioinformatics Tutorials
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="Whole-genome de-novo Assembly" class="md-nav__link">
      Whole-genome de-novo Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome Assembly" class="md-nav__link">
      Metagenome Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metabarcoding/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-genome analysis" class="md-nav__link">
      Pan-genome analysis
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../qc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        RNA-Seq analysis
      </label>
    
    <a href="./" title="RNA-Seq analysis" class="md-nav__link md-nav__link--active">
      RNA-Seq analysis
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#load-salmon" title="Load salmon" class="md-nav__link">
    Load salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indexing-transcriptome" title="Indexing transcriptome" class="md-nav__link">
    Indexing transcriptome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quantify-reads-using-salmon" title="Quantify reads using salmon" class="md-nav__link">
    Quantify reads using salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-read-counts-using-tximport" title="Import read counts using tximport" class="md-nav__link">
    Import read counts using tximport
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-using-deseq2" title="Differential expression using DESeq2" class="md-nav__link">
    Differential expression using DESeq2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#kegg-pathway-analysis" title="KEGG pathway analysis" class="md-nav__link">
    KEGG pathway analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#thanks" title="Thanks" class="md-nav__link">
    Thanks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../unix/" title="Unix" class="md-nav__link">
      Unix
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variant_calling/" title="Variant calling" class="md-nav__link">
      Variant calling
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../wms/" title="Wms" class="md-nav__link">
      Wms
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#load-salmon" title="Load salmon" class="md-nav__link">
    Load salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#indexing-transcriptome" title="Indexing transcriptome" class="md-nav__link">
    Indexing transcriptome
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quantify-reads-using-salmon" title="Quantify reads using salmon" class="md-nav__link">
    Quantify reads using salmon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#import-read-counts-using-tximport" title="Import read counts using tximport" class="md-nav__link">
    Import read counts using tximport
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#differential-expression-using-deseq2" title="Differential expression using DESeq2" class="md-nav__link">
    Differential expression using DESeq2
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#kegg-pathway-analysis" title="KEGG pathway analysis" class="md-nav__link">
    KEGG pathway analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#thanks" title="Thanks" class="md-nav__link">
    Thanks
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="rna-seq-analysis">RNA-Seq analysis</h1>
<h2 id="load-salmon">Load salmon</h2>
<pre><code>module load Salmon
</code></pre>

<h2 id="downloading-the-data">Downloading the data</h2>
<p>For this tutorial we will use the test data from <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004393">this</a> paper:</p>
<blockquote>
<p>Malachi Griffith<em>, Jason R. Walker, Nicholas C. Spies, Benjamin J. Ainscough, Obi L. Griffith</em>. 2015. Informatics for RNA-seq: A web resource for analysis on the cloud. PLoS Comp Biol. 11(8):e1004393.</p>
</blockquote>
<p>The test data consists of two commercially available RNA samples: Universal Human Reference (UHR) and Human Brain Reference (HBR). The UHR is total RNA isolated from a diverse set of 10 cancer cell lines. The HBR is total RNA isolated from the brains of 23 Caucasians, male and female, of varying age but mostly 60-80 years old.</p>
<p>In addition, a spike-in control was used. Specifically we added an aliquot of the ERCC ExFold RNA Spike-In Control Mixes to each sample. The spike-in consists of 92 transcripts that are present in known concentrations across a wide abundance range (from very few copies to many copies). This range allows us to test the degree to which the RNA-seq assay (including all laboratory and analysis steps) accurately reflects the relative abundance of transcript species within a sample. There are two 'mixes' of these transcripts to allow an assessment of differential expression output between samples if you put one mix in each of your two comparisons. In our case, Mix1 was added to the UHR sample, and Mix2 was added to the HBR sample. We also have 3 complete experimental replicates for each sample. This allows us to assess the technical variability of our overall process of producing RNA-seq data in the lab.</p>
<p>For all libraries we prepared low-throughput (Set A) TruSeq Stranded Total RNA Sample Prep Kit libraries with Ribo-Zero Gold to remove both cytoplasmic and mitochondrial rRNA. Triplicate, indexed libraries were made starting with 100ng Agilent/Strategene Universal Human Reference total RNA and 100ng Ambion Human Brain Reference total RNA. The Universal Human Reference replicates received 2 ul of 1:1000 ERCC Mix 1. The Human Brain Reference replicates received 1:1000 ERCC Mix 2. The libraries were quantified with KAPA Library Quantification qPCR and adjusted to the appropriate concentration for sequencing. The triplicate, indexed libraries were then pooled prior to sequencing. Each pool of three replicate libraries were sequenced across 2 lanes of a HiSeq 2000 using paired-end sequence chemistry with 100bp read lengths.</p>
<p>So to summarize we have:</p>
<ul>
<li>UHR + ERCC Spike-In Mix1, Replicate 1</li>
<li>UHR + ERCC Spike-In Mix1, Replicate 2</li>
<li>UHR + ERCC Spike-In Mix1, Replicate 3</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 1</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 2</li>
<li>HBR + ERCC Spike-In Mix2, Replicate 3</li>
</ul>
<p>You can download the data from <a href="http://139.162.178.46/files/tutorials/toy_rna.tar.gz">here</a>.</p>
<p>Unpack the data and go into the toy_rna directory</p>
<pre><code>tar xzf toy_rna.tar.gz
cd toy_rna
</code></pre>

<h2 id="indexing-transcriptome">Indexing transcriptome</h2>
<pre><code>salmon index -t chr22_transcripts.fa -i chr22_index
</code></pre>

<h2 id="quantify-reads-using-salmon">Quantify reads using salmon</h2>
<pre><code class="bash">for i in *_R1.fastq.gz
do
   prefix=$(basename $i _R1.fastq.gz)
   salmon quant -i chr22_index --libType A \
          -1 ${prefix}_R1.fastq.gz -2 ${prefix}_R2.fastq.gz -o quant/${prefix};
done
</code></pre>

<p>This loop simply goes through each sample and invokes salmon using fairly basic options:</p>
<ul>
<li>The -i argument tells salmon where to find the index</li>
<li>--libType A tells salmon that it should automatically determine the library type of the sequencing reads (e.g. stranded vs. unstranded etc.)</li>
<li>The -1 and -2 arguments tell salmon where to find the left and right reads for this sample (notice, salmon will accept gzipped FASTQ files directly).</li>
<li>the -o argument specifies the directory where salmon’s quantification results sould be written.</li>
</ul>
<p>Salmon exposes many different options to the user that enable extra features or modify default behavior. However, the purpose and behavior of all of those options is beyond the scope of this introductory tutorial. You can read about salmon’s many options in the <a href="http://salmon.readthedocs.io/en/latest/">documentation</a>.</p>
<p>After the salmon commands finish running, you should have a directory named <code>quant</code>, which will have a sub-directory for each sample. These sub-directories contain the quantification results of salmon, as well as a lot of other information salmon records about the sample and the run. The main output file (called quant.sf) is rather self-explanatory. For example, take a peek at the quantification file for sample <code>HBR_Rep1</code> in <code>quant/HBR_Rep1/quant.sf</code> and you’ll see a simple TSV format file listing the name (Name) of each transcript, its length (Length), effective length (EffectiveLength) (more details on this in the documentation), and its abundance in terms of Transcripts Per Million (TPM) and estimated number of reads (NumReads) originating from this transcript.</p>
<h2 id="import-read-counts-using-tximport">Import read counts using tximport</h2>
<p>Using the tximport R package, you can import salmon’s transcript-level quantifications and optionally aggregate them to the gene level for gene-level differential expression analysis. </p>
<p>First, open up your favourite R IDE and install the necessary packages:</p>
<pre><code class="R">source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;tximport&quot;)
biocLite(&quot;GenomicFeatures&quot;)

install.packages(&quot;readr&quot;)
</code></pre>

<p>Then load the modules:</p>
<pre><code class="R">library(tximport)
library(GenomicFeatures)
library(readr)
</code></pre>

<p>Salmon did the quantifiation of the transcript level. We want to see which genes are differentially expressed, so we need to link the transcript names to the gene names. We can use our .gtf annotation for that, and the GenomicFeatures package:</p>
<pre><code class="R">txdb &lt;- makeTxDbFromGFF(&quot;chr22_genes.gtf&quot;)
k &lt;- keys(txdb, keytype = &quot;GENEID&quot;)
tx2gene &lt;- select(txdb, keys = k, keytype = &quot;GENEID&quot;, columns = &quot;TXNAME&quot;)
head(tx2gene)
</code></pre>

<p>Now we can import the salmon quantification. First, download the file with sample descriptions from <a href="https://raw.githubusercontent.com/HadrienG/tutorials/master/data/samples.txt">here</a> and put it in the toy_rna directory. Then, use that file to load the corresponding quantification data.</p>
<pre><code class="R">samples &lt;- read.table(&quot;samples.txt&quot;, header = TRUE)
files &lt;- file.path(&quot;quant&quot;, samples$sample, &quot;quant.sf&quot;)
names(files) &lt;- paste0(samples$sample)
txi.salmon &lt;- tximport(files, type = &quot;salmon&quot;, tx2gene = tx2gene, reader = read_tsv)
</code></pre>

<p>Take a look at the data:</p>
<pre><code class="R">head(txi.salmon$counts)
</code></pre>

<h2 id="differential-expression-using-deseq2">Differential expression using DESeq2</h2>
<p>Install the necessary package:</p>
<pre><code class="R">biocLite('DESeq2')
</code></pre>

<p>Then load it:</p>
<pre><code class="R">library(DESeq2)
</code></pre>

<p>Instantiate the DESeqDataSet and generate result table. See <code>?DESeqDataSetFromTximport</code> and <code>?DESeq</code> for more information about the steps performed by the program.</p>
<pre><code class="R">dds &lt;- DESeqDataSetFromTximport(txi.salmon, samples, ~condition)
dds &lt;- DESeq(dds)
res &lt;- results(dds)
</code></pre>

<p>Run the <code>summary</code> command to get an idea of how many genes are up- and downregulated between the two conditions:</p>
<p><code>summary(res)</code></p>
<p>DESeq uses a negative binomial distribution. Such distributions have two parameters: mean and dispersion. The dispersion is a parameter describing how much the variance deviates from the mean.</p>
<p>You can read more about the methods used by DESeq2 in the <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">paper</a> or the <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/DESeq/inst/doc/DESeq.pdf">vignette</a></p>
<p>Plot dispersions:</p>
<pre><code class="R">plotDispEsts(dds, main=&quot;Dispersion plot&quot;)
</code></pre>

<p>For clustering and heatmaps, we need to log transform our data:</p>
<pre><code class="R">rld &lt;- rlogTransformation(dds)
head(assay(rld))
</code></pre>

<p>Then, we create a sample distance heatmap:</p>
<pre><code class="R">library(RColorBrewer)
library(gplots) # you may need to install this package

(mycols &lt;- brewer.pal(8, &quot;Dark2&quot;)[1:length(unique(samples$condition))])
sampleDists &lt;- as.matrix(dist(t(assay(rld))))
heatmap.2(as.matrix(sampleDists), key=F, trace=&quot;none&quot;,
          col=colorpanel(100, &quot;black&quot;, &quot;white&quot;),
          ColSideColors=mycols[samples$condition],
          RowSideColors=mycols[samples$condition],
          margin=c(10, 10), main=&quot;Sample Distance Matrix&quot;)
</code></pre>

<p>We can also plot a PCA:</p>
<pre><code class="R">DESeq2::plotPCA(rld, intgroup=&quot;condition&quot;)
</code></pre>

<p>It is time to look at some p-values:</p>
<pre><code class="R">table(res$padj&lt;0.05)
res &lt;- res[order(res$padj), ]
resdata &lt;- merge(as.data.frame(res), as.data.frame(counts(dds, normalized=TRUE)), by=&quot;row.names&quot;, sort=FALSE)
names(resdata)[1] &lt;- &quot;Gene&quot;
head(resdata)
</code></pre>

<p>Examine plot of p-values, the MA plot and the Volcano Plot:</p>
<pre><code class="R">hist(res$pvalue, breaks=50, col=&quot;grey&quot;)
DESeq2::plotMA(dds, ylim=c(-1,1), cex=1)

# Volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main=&quot;Volcano plot&quot;, xlim=c(-2.5,2)))
with(subset(res, padj&lt;.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col=&quot;red&quot;))
</code></pre>

<h2 id="kegg-pathway-analysis">KEGG pathway analysis</h2>
<p>As always, install and load the necessary packages:</p>
<pre><code class="R">biocLite(&quot;AnnotationDbi&quot;)
biocLite(&quot;org.Hs.eg.db&quot;)
biocLite(&quot;pathview&quot;)
biocLite(&quot;gage&quot;)
biocLite(&quot;gageData&quot;)

library(AnnotationDbi)
library(org.Hs.eg.db)
library(pathview)
library(gage)
library(gageData)
</code></pre>

<p>Let’s use the <code>mapIds</code> function to add more columns to the results. The row.names of our results table has the Ensembl gene ID (our key), so we need to specify  <code>keytype=ENSEMBL</code>. The column argument tells the <code>mapIds</code> function which information we want, and the <code>multiVals</code> argument tells the function what to do if there are multiple possible values for a single input value. Here we ask to just give us back the first one that occurs in the database. Let’s get the Entrez IDs, gene symbols, and full gene names.</p>
<pre><code class="R">res$symbol &lt;- mapIds(org.Hs.eg.db,
                     keys=row.names(res),
                     column=&quot;SYMBOL&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)
res$entrez &lt;- mapIds(org.Hs.eg.db,
                     keys=row.names(res),
                     column=&quot;ENTREZID&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)
res$name &lt;- mapIds(org.Hs.eg.db,
                     keys=row.names(res),
                     column=&quot;GENENAME&quot;,
                     keytype=&quot;ENSEMBL&quot;,
                     multiVals=&quot;first&quot;)

head(res)
</code></pre>

<p>We’re going to use the <a href="https://bioconductor.org/packages/release/bioc/html/gage.html">gage</a> package for pathway analysis, and the <a href="https://bioconductor.org/packages/release/bioc/html/pathview.html">pathview</a> package to draw a pathway diagram.</p>
<p>The gageData package has pre-compiled databases mapping genes to KEGG pathways and GO terms for common organisms:</p>
<pre><code class="R">data(kegg.sets.hs)
data(sigmet.idx.hs)
kegg.sets.hs &lt;- kegg.sets.hs[sigmet.idx.hs]
head(kegg.sets.hs, 3)
</code></pre>

<p>Run the pathway analysis. See help on the gage function with <code>?gage</code>. Specifically, you might want to try changing the value of same.dir.</p>
<pre><code class="R">foldchanges &lt;- res$log2FoldChange
names(foldchanges) &lt;- res$entrez
keggres &lt;- gage(foldchanges, gsets=kegg.sets.hs, same.dir=TRUE)
lapply(keggres, head)
</code></pre>

<p>Pull out the top 5 upregulated pathways, then further process that just to get the IDs. We’ll use these KEGG pathway IDs downstream for plotting. The <code>dplyr</code> package is required to use the pipe (<code>%&gt;%</code>) construct. </p>
<pre><code class="R">library(dplyr)

# Get the pathways
keggrespathways &lt;- data.frame(id=rownames(keggres$greater), keggres$greater) %&gt;%
  tbl_df() %&gt;%
  filter(row_number()&lt;=5) %&gt;%
  .$id %&gt;%
  as.character()
keggrespathways

# Get the IDs.
keggresids &lt;- substr(keggrespathways, start=1, stop=8)
keggresids
</code></pre>

<p>Finally, the <code>pathview()</code> function in the pathview package makes the plots. Let’s write a function so we can loop through and draw plots for the top 5 pathways we created above.</p>
<pre><code class="R"># Define plotting function for applying later
plot_pathway &lt;- function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species=&quot;hsa&quot;, new.signature=FALSE)

# Unload dplyr since it conflicts with the next line
detach(&quot;package:dplyr&quot;, unload=T)

# plot multiple pathways (plots saved to disk and returns a throwaway list object)
tmp &lt;- sapply(keggresids, function(pid) pathview(gene.data=foldchanges, pathway.id=pid, species=&quot;hsa&quot;))
</code></pre>

<h4 id="thanks">Thanks</h4>
<p>This material was inspired by Stephen Turner's blog post:</p>
<blockquote>
<p>Tutorial: RNA-seq differential expression &amp; pathway analysis with Sailfish, DESeq2, GAGE, and Pathview: http://www.gettinggeneticsdone.com/2015/12/tutorial-rna-seq-differential.html</p>
</blockquote>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../qc/" title="Quality control" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Quality control
              </span>
            </div>
          </a>
        
        
          <a href="../unix/" title="Unix" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Unix
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application-8e4952e681.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:".."}})</script>
      
    
    
      
    
  </body>
</html>