



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.0.4">
    
    
      
        <title>Whole Metagenome Sequencing - Bioinformatics Walkthroughs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application-709eec9410.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      <script src="../assets/javascripts/modernizr-e826f8942a.js"></script>
    
    
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    
    
    
      <body data-md-color-primary="pink" data-md-color-accent="pink">
    
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="Bioinformatics Walkthroughs" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
            
            Whole Metagenome Sequencing
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Bioinformatics Walkthroughs
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/HadrienG/tutorials/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      HadrienG/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../command_line/" title="The command-line" class="md-nav__link">
      The command-line
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../qc/" title="Quality Control and Trimming" class="md-nav__link">
      Quality Control and Trimming
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../mapping/" title="Mapping and Variant Calling" class="md-nav__link">
      Mapping and Variant Calling
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="De-novo Genome Assembly" class="md-nav__link">
      De-novo Genome Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-Genome Analysis" class="md-nav__link">
      Pan-Genome Analysis
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../16S/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Whole Metagenome Sequencing
      </label>
    
    <a href="./" title="Whole Metagenome Sequencing" class="md-nav__link md-nav__link--active">
      Whole Metagenome Sequencing
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" title="Table of Contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#microbiome-used" title="Microbiome used" class="md-nav__link">
    Microbiome used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#whole-metagenome-sequencing" title="Whole Metagenome Sequencing" class="md-nav__link">
    Whole Metagenome Sequencing
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#softwares-required-for-this-tutorial" title="Softwares Required for this Tutorial" class="md-nav__link">
    Softwares Required for this Tutorial
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-the-data-and-checking-their-quality" title="Getting the Data and Checking their Quality" class="md-nav__link">
    Getting the Data and Checking their Quality
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-classification" title="Taxonomic Classification" class="md-nav__link">
    Taxonomic Classification
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#abundance-estimation-using-bracken" title="Abundance estimation using Bracken" class="md-nav__link">
    Abundance estimation using Bracken
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization" title="Visualization" class="md-nav__link">
    Visualization
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alternative-1-pavian" title="Alternative 1: Pavian" class="md-nav__link">
    Alternative 1: Pavian
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome assembly" class="md-nav__link">
      Metagenome assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#table-of-contents" title="Table of Contents" class="md-nav__link">
    Table of Contents
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#microbiome-used" title="Microbiome used" class="md-nav__link">
    Microbiome used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#whole-metagenome-sequencing" title="Whole Metagenome Sequencing" class="md-nav__link">
    Whole Metagenome Sequencing
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#softwares-required-for-this-tutorial" title="Softwares Required for this Tutorial" class="md-nav__link">
    Softwares Required for this Tutorial
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-the-data-and-checking-their-quality" title="Getting the Data and Checking their Quality" class="md-nav__link">
    Getting the Data and Checking their Quality
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-classification" title="Taxonomic Classification" class="md-nav__link">
    Taxonomic Classification
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#abundance-estimation-using-bracken" title="Abundance estimation using Bracken" class="md-nav__link">
    Abundance estimation using Bracken
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#visualization" title="Visualization" class="md-nav__link">
    Visualization
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#alternative-1-pavian" title="Alternative 1: Pavian" class="md-nav__link">
    Alternative 1: Pavian
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/HadrienG/tutorials/edit/master/docs/wms.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="whole-metagenome-sequencin">Whole Metagenome Sequencin<a class="headerlink" href="#whole-metagenome-sequencin" title="Permanent link">&para;</a></h1>
<h3 id="table-of-contents">Table of Contents<a class="headerlink" href="#table-of-contents" title="Permanent link">&para;</a></h3>
<ul>
<li><a href="#introduction">Introduction</a><ul>
<li><a href="#the-pig-microbiome">The Pig Microbiome</a></li>
<li><a href="#whole-metagenome-sequencing">Whole Metagenome Sequencing</a></li>
</ul>
</li>
<li><a href="#softwares-required-for-this-tutorial">Softwares Required for this Tutorial</a></li>
<li><a href="#getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality</a></li>
<li><a href="#taxonomic-classification">Taxonomic Classification</a></li>
<li><a href="#visualization">Visualization</a></li>
</ul>
<h3 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h3>
<h4 id="microbiome-used">Microbiome used<a class="headerlink" href="#microbiome-used" title="Permanent link">&para;</a></h4>
<p>In this tutorial we will compare samples from the Pig Gut Microbiome to samples from the Human Gut Microbiome. Below you'll find a brief description of the two projects:</p>
<p>The Pig Microbiome:</p>
<blockquote>
<p>Pig is a main species for livestock and biomedicine. The pig genome sequence was recently reported. To boost research, we established a catalogue of the genes of the gut microbiome based on faecal samples of 287 pigs from France, Denmark and China. More than 7.6 million non-redundant genes representing 719 metagenomic species were identified by deep metagenome sequencing, highlighting more similarities with the human than with the mouse catalogue. The pig and human catalogues share only 12.6 and 9.3 % of their genes, respectively, but 70 and 95% of their functional pathways. The pig gut microbiota is influenced by gender, age and breed. Analysis of the prevalence of antibiotics resistance genes (ARGs) reflected antibiotics supplementation in each farm system, and revealed that non-antibiotics-fed animals still harbour ARGs. The pig catalogue creates a resource for whole metagenomics-based studies, highly valuable for research in biomedicine and for sustainable knowledge-based pig farming</p>
</blockquote>
<p>The Human Microbiome:</p>
<blockquote>
<p>We are facing a global metabolic health crisis provoked by an obesity epidemic. Here we report the human gut microbial composition in a population sample of 123 non-obese and 169 obese Danish individuals. We find two groups of individuals that differ by the number of gut microbial genes and thus gut bacterial richness. They harbour known and previously unknown bacterial species at different proportions; individuals with a low bacterial richness (23% of the population) are characterized by more marked overall adiposity, insulin resistance and dyslipidaemia and a more pronounced inflammatory phenotype when compared with high bacterial richness individuals. The obese individuals among the former also gain more weight over time. Only a few bacterial species are sufficient to distinguish between individuals with high and low bacterial richness, and even between lean and obese. Our classifications based on variation in the gut microbiome identify subsets of individuals in the general white adult population who may be at increased risk of progressing to adiposity-associated co-morbidities</p>
</blockquote>
<h4 id="whole-metagenome-sequencing">Whole Metagenome Sequencing<a class="headerlink" href="#whole-metagenome-sequencing" title="Permanent link">&para;</a></h4>
<p>Whole Metagenome sequencing (WMS), or shotgun metagenome sequencing, is a relatively new and powerful sequencing approach that provides insight into community biodiversity and function. On the contrary of Metabarcoding, where only a specific region of the bacterial community (the 16s rRNA) is sequenced, WMS aims at sequencing all the genomic material present in the environment.</p>
<p>The choice of shotgun or 16S approaches is usually dictated by the nature of the studies being conducted. For instance, 16S is well suited for analysis of large number of samples, i.e., multiple patients, longitudinal studies, etc. but offers limited taxonomical and functional resolution. WMS is generally more expensive but offers increased resolution, and allows the discovery of viruses as well as other mobile genetic elements.</p>
<h3 id="softwares-required-for-this-tutorial">Softwares Required for this Tutorial<a class="headerlink" href="#softwares-required-for-this-tutorial" title="Permanent link">&para;</a></h3>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></li>
<li><a href="https://ccb.jhu.edu/software/kraken/">Kraken</a></li>
<li><a href="https://ccb.jhu.edu/software/bracken/">Bracken</a></li>
<li><a href="https://www.r-project.org/">R</a></li>
<li><a href="https://github.com/fbreitwieser/pavian">Pavian</a></li>
</ul>
<h3 id="getting-the-data-and-checking-their-quality">Getting the Data and Checking their Quality<a class="headerlink" href="#getting-the-data-and-checking-their-quality" title="Permanent link">&para;</a></h3>
<p>If you are reading this tutorial online and haven't cloned the directory, first download and unpack the data:</p>
<pre><code>wget http://77.235.253.14/metlab/wms.tar
tar xvf wms.tar
cd wms
</code></pre>

<p>We'll use FastQC to check the quality of our data. FastQC can be downloaded and
run on a Windows or Linux computer without installation. It is available <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here</a></p>
<p>Start FastQC and select the fastq file you just downloaded with <code>file -&gt; open</code><br />
What do you think about the quality of the reads? Do they need trimming? Are there still adapters
present? Overrepresented sequences?</p>
<p>Alternatively, run fastqc on the command-line:</p>
<p><code>fastqc *.fastq.gz</code></p>
<p>If the quality appears to be good, it's because it was probably the cleaned reads that were deposited into SRA.
We can directly move to the classification step.</p>
<h3 id="taxonomic-classification">Taxonomic Classification<a class="headerlink" href="#taxonomic-classification" title="Permanent link">&para;</a></h3>
<p><a href="https://ccb.jhu.edu/software/kraken/">Kraken</a> is a system for assigning taxonomic labels to short DNA sequences (i.e. reads)<br />
Kraken aims to achieve high sensitivity and high speed by utilizing exact alignments of k-mers and a novel classification algorithm (sic).</p>
<p>In short, kraken uses a new approach with exact k-mer matching to assign taxonomy to short reads. It is <em>extremely</em> fast compared to traditional
approaches (i.e. BLAST).</p>
<p>By default, the authors of kraken built their database based on RefSeq Bacteria, Archea and Viruses. We'll use it for the purpose of this tutorial.</p>
<p><strong>NOTE: The database may have been installed already! Ask your instructor!</strong></p>
<pre><code class="bash"># You might not need this step (for example if you're working on Uppmax!)
wget https://ccb.jhu.edu/software/kraken/dl/minikraken.tgz
tar xzf minikraken.tgz
$KRAKEN_DB=minikraken_20141208
</code></pre>

<p>Now run kraken on the reads</p>
<pre><code class="bash">mkdir kraken_results
for i in *_1.fastq.gz
do
    prefix=$(basename $i _1.fastq.gz)
    # set number of threads to number of cores if running under SLURM, otherwise use 2 threads
    nthreads=${SLURM_NPROCS:=2}
    kraken --db $KRAKEN_DB --threads ${nthreads} --fastq-input --gzip-compressed \
        ${prefix}_1.fastq.gz ${prefix}_2.fastq.gz &gt; kraken_results/${prefix}.tab
    kraken-report --db $KRAKEN_DB \
        kraken_results/${prefix}.tab &gt; kraken_results/${prefix}_tax.txt
done
</code></pre>

<p>which produces a tab-delimited file with an assigned TaxID for each read.</p>
<p>Kraken includes a script called <code>kraken-report</code> to transform this file into a "tree" view with the percentage of reads assigned to each taxa. We've run this script at each step in the loop. Take a look at the <code>_tax.txt</code> files!</p>
<h3 id="abundance-estimation-using-bracken">Abundance estimation using Bracken<a class="headerlink" href="#abundance-estimation-using-bracken" title="Permanent link">&para;</a></h3>
<p>Bracken (Bayesian Reestimation of Abundance with KrakEN) is a highly accurate statistical method that computes the abundance of species in DNA sequences from a metagenomics sample</p>
<p>Before starting, you need to install Bracken:</p>
<pre><code class="bash">cd
git clone https://github.com/jenniferlu717/Bracken.git
chmod 755 Bracken/*.py
chmod 755 Bracken/*.pl
export PATH=$PATH:$HOME/Bracken
</code></pre>

<p>Unfortunately, Uppmax lacks some perl packages necessary for Bracken to work:</p>
<p>Follow the tutorial <a href="http://www.uppmax.uu.se/support/faq/software-faq/installing-local-perl-packages/">here</a> to install <code>cpanm</code></p>
<p>then install the two perl libraries that are missing:</p>
<pre><code class="bash">cpanm Parallel::ForkManager
cpanm List::MoreUtils
</code></pre>

<p>Three steps are necessary to set up Kraken abundance estimation.</p>
<ol>
<li>
<p>Classify all reads using Kraken and Generate a Kraken report file. We've done this!</p>
</li>
<li>
<p>Search all library input sequences against the database and compute the classifications for each perfect read of ${READ_LENGTH} base pairs from one of the input sequences.</p>
</li>
</ol>
<pre><code class="bash">find -L $KRAKEN_DB/library -name &quot;*.fna&quot; -o -name &quot;*.fa&quot; -o -name &quot;*.fasta&quot; &gt; genomes.list
cat $(grep -v '^#' genomes.list) &gt; genomes.fasta
kraken --db=${KRAKEN_DB} --fasta-input --threads=${SLURM_NPROCS:=10} kraken.fasta &gt; database.kraken
count-kmer-abundances.pl --db=${KRAKEN_DB} --read-length=100 database.kraken &gt; database100mers.kraken_cnts
</code></pre>

<ol>
<li>Generate the kmer distribution file</li>
</ol>
<pre><code class="bash">python generate_kmer_distribution.py -i database100mers.kraken_cnts -o KMER_DISTR.TXT
</code></pre>

<p>Now, given the expected kmer distribution for genomes in a kraken database along
with a kraken report file, the number of reads belonging to each species (or
genus) is estimated using the estimate_abundance.py file, run with the
following command line:</p>
<p><code>python estimate_abundance.py -i KRAKEN.REPORT -k KMER_DISTR.TXT -o OUTPUT_FILE.TXT</code></p>
<p>Run this command for the six <code>_tax.txt</code> files that you generated with kraken!</p>
<p>The following required parameters must be specified:
- KRAKEN.REPORT     :: the kraken report generated for a given dataset
- KMER_DISTR.TXT    :: the file generated by generate_kmer_distribution.py
- OUTPUT_FILE.TXT   :: the desired name of the output file to be generated by the code</p>
<h3 id="visualization">Visualization<a class="headerlink" href="#visualization" title="Permanent link">&para;</a></h3>
<h4 id="alternative-1-pavian">Alternative 1: Pavian<a class="headerlink" href="#alternative-1-pavian" title="Permanent link">&para;</a></h4>
<p>Pavian is a web application for exploring metagenomics classification results.</p>
<p>Install and run Pavian:</p>
<p>(In R or Rstudio)</p>
<pre><code class="R">## Installs required packages from CRAN and Bioconductor
source(&quot;https://raw.githubusercontent.com/fbreitwieser/pavian/master/inst/shinyapp/install-pavian.R&quot;)
pavian::runApp(port=5000)
</code></pre>

<p>Pavian will be available at http://127.0.0.1:5000</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../16S/" title="Metabarcoding" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metabarcoding
              </span>
            </div>
          </a>
        
        
          <a href="../meta_assembly/" title="Metagenome assembly" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Metagenome assembly
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://github.com/HadrienG" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://twitter.com/HGourle" class="md-footer-social__link fa fa-twitter"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application-8e4952e681.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:".."}})</script>
      
    
    
      
    
  </body>
</html>