



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.0.4">
    
    
      
        <title>Pan-genome analysis - SGBC Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application-709eec9410.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      <script src="../assets/javascripts/modernizr-e826f8942a.js"></script>
    
    
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    
    
    
      <body data-md-color-primary="pink" data-md-color-accent="pink">
    
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="SGBC Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
            
            Pan-genome analysis
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    SGBC Bioinformatics Tutorials
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="Whole-genome de-novo Assembly" class="md-nav__link">
      Whole-genome de-novo Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome Assembly" class="md-nav__link">
      Metagenome Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metabarcoding/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Pan-genome analysis
      </label>
    
    <a href="./" title="Pan-genome analysis" class="md-nav__link md-nav__link--active">
      Pan-genome analysis
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-data" title="Getting the data" class="md-nav__link">
    Getting the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation-of-the-genomes" title="Annotation of the genomes" class="md-nav__link">
    Annotation of the genomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pan-genome-analysis_1" title="Pan-genome analysis" class="md-nav__link">
    Pan-genome analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-the-result" title="Plotting the result" class="md-nav__link">
    Plotting the result
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../qc/" title="Qc" class="md-nav__link">
      Qc
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna_seq/" title="Rna seq" class="md-nav__link">
      Rna seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../unix/" title="Unix" class="md-nav__link">
      Unix
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variant_calling/" title="Variant calling" class="md-nav__link">
      Variant calling
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../wms/" title="Wms" class="md-nav__link">
      Wms
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-data" title="Getting the data" class="md-nav__link">
    Getting the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#annotation-of-the-genomes" title="Annotation of the genomes" class="md-nav__link">
    Annotation of the genomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pan-genome-analysis_1" title="Pan-genome analysis" class="md-nav__link">
    Pan-genome analysis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#plotting-the-result" title="Plotting the result" class="md-nav__link">
    Plotting the result
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="pan-genome-analysis">Pan-genome analysis</h1>
<p>In this tutorial we will learn how to determine a pan-genome from a collection of isolate genomes.</p>
<p>This tutorial is inspired from <a href="https://github.com/microgenomics/tutorials/blob/master/pangenome.md">Genome annotation and Pangenome Analysis</a> from the CBIB in Santiago, Chile</p>
<h2 id="getting-the-data">Getting the data</h2>
<p>We'll use six <em>Listeria monocytogenes</em> genomes in this tutorial.</p>
<pre><code class="bash">wget https://github.com/HadrienG/tutorials/blob/master/data/pangenome.tar.gz
tar xzf pangenome.tar.gz
cd pangenome
</code></pre>

<p>These genomes correspond to isolates of <em>L. monocytogenes</em> reported in</p>
<blockquote>
<p>Xiangyu Deng, Adam M Phillippy, Zengxin Li, Steven L Salzberg and Wei Zhang. (2010) Probing the pan-genome of Listeria monocytogenes: new insights into intraspecific niche expansion and genomic diversification. doi:10.1186/1471-2164-11-500</p>
</blockquote>
<p>The six genomes you downloaded were selected based on their level of completeness (finished; contigs, etc) and their genotype (type I-III):</p>
<table>
<thead>
<tr>
<th>Genome Assembly</th>
<th>Genome Accession</th>
<th>Genotype</th>
<th>Sequenced by</th>
<th>Status</th>
</tr>
</thead>
<tbody>
<tr>
<td>GCA_000026705</td>
<td>FM242711</td>
<td>type I</td>
<td>Institut_Pasteur</td>
<td>Finished</td>
</tr>
<tr>
<td>GCA_000008285</td>
<td>AE017262</td>
<td>type I</td>
<td>TIGR</td>
<td>Finished</td>
</tr>
<tr>
<td>GCA_000168815</td>
<td>AATL00000000</td>
<td>type I</td>
<td>Broad Institute</td>
<td>79 contigs</td>
</tr>
<tr>
<td>GCA_000196035</td>
<td>AL591824</td>
<td>type II</td>
<td>European Consortium</td>
<td>Finished</td>
</tr>
<tr>
<td>GCA_000168635</td>
<td>AARW00000000</td>
<td>type II</td>
<td>Broad Institute</td>
<td>25 contigs</td>
</tr>
<tr>
<td>GCA_000021185</td>
<td>CP001175</td>
<td>type III</td>
<td>MSU</td>
<td>Finished</td>
</tr>
</tbody>
</table>
<h2 id="annotation-of-the-genomes">Annotation of the genomes</h2>
<p>By annotating the genomes we mean to add information regarding genes, their location, strandedness, and features and attributes. Now that you have the genomes, we need to annotate them to determine the location and attributes of the genes contained in them. We will use Prokka for the annotation.</p>
<pre><code>prokka --kingdom Bacteria --outdir prokka_GCA_000008285 --genus Listeria --locustag GCA_000008285 GCA_000008285.1_ASM828v1_genomic.fna
</code></pre>

<p>Annotate the 6 genomes, by replacing the <code>-outdir</code> and <code>-locustag</code> and <code>fasta file</code> accordingly.</p>
<h2 id="pan-genome-analysis_1">Pan-genome analysis</h2>
<p>put all the .gff files in the same folder (e.g., ./gff) and run Roary</p>
<p><code>roary -f results -e -n -v gff/*.gff</code></p>
<p>Roary will get all the coding sequences, convert them into protein, and create pre-clusters. Then, using BLASTP and MCL, Roary will create clusters, and check for paralogs. Finally, Roary will take every isolate and order them by presence/absence of orthologs. The summary output is present in the <code>summary_statistics.txt</code> file.</p>
<p>Additionally, Roary produces a <code>gene_presence_absence.csv</code> file that can be opened in any spreadsheet software to manually explore the results. In this file, you will find information such as gene name and gene annotation, and, of course, whether a gene is present in a genome or not.</p>
<h2 id="plotting-the-result">Plotting the result</h2>
<p>Roary comes with a python script that allows you to generate a few plots to graphically assess your analysis output.</p>
<p>First, we need to generate a tree file from the alignment generated by Roary:</p>
<pre><code>FastTree –nt –gtr core_gene_alignment.aln &gt; my_tree.newick
</code></pre>

<p>Then we can plot the Roary results:</p>
<pre><code>roary_plots.py my_tree.newick gene_presence_absence.csv
</code></pre>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../metabarcoding/" title="Metabarcoding" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metabarcoding
              </span>
            </div>
          </a>
        
        
          <a href="../qc/" title="Qc" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Qc
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application-8e4952e681.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:".."}})</script>
      
    
    
      
    
  </body>
</html>