



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.0.4">
    
    
      
        <title>Quality control - SGBC Bioinformatics Tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/application-709eec9410.css">
      
        <link rel="stylesheet" href="../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      <script src="../assets/javascripts/modernizr-e826f8942a.js"></script>
    
    
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    
    
    
      <body data-md-color-primary="pink" data-md-color-accent="pink">
    
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href=".." title="SGBC Bioinformatics Tutorials" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
            
            Quality control
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    SGBC Bioinformatics Tutorials
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href=".." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../annotation/" title="Genome Annotation" class="md-nav__link">
      Genome Annotation
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../assembly/" title="Whole-genome de-novo Assembly" class="md-nav__link">
      Whole-genome de-novo Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../file_formats/" title="File Formats" class="md-nav__link">
      File Formats
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../meta_assembly/" title="Metagenome Assembly" class="md-nav__link">
      Metagenome Assembly
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../metabarcoding/" title="Metabarcoding" class="md-nav__link">
      Metabarcoding
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../pan_genome/" title="Pan-genome analysis" class="md-nav__link">
      Pan-genome analysis
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Quality control
      </label>
    
    <a href="./" title="Quality control" class="md-nav__link md-nav__link--active">
      Quality control
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc" title="FastQC" class="md-nav__link">
    FastQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scythe" title="Scythe" class="md-nav__link">
    Scythe
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sickle" title="Sickle" class="md-nav__link">
    Sickle
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extra-exercises" title="Extra exercises" class="md-nav__link">
    Extra exercises
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../rna_seq/" title="Rna seq" class="md-nav__link">
      Rna seq
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../unix/" title="Unix" class="md-nav__link">
      Unix
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../variant_calling/" title="Variant calling" class="md-nav__link">
      Variant calling
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../wms/" title="Wms" class="md-nav__link">
      Wms
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#downloading-the-data" title="Downloading the data" class="md-nav__link">
    Downloading the data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fastqc" title="FastQC" class="md-nav__link">
    FastQC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#scythe" title="Scythe" class="md-nav__link">
    Scythe
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sickle" title="Sickle" class="md-nav__link">
    Sickle
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#extra-exercises" title="Extra exercises" class="md-nav__link">
    Extra exercises
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="quality-control">Quality control</h1>
<p>In this practical you will learn to import, view and check the quality of NGS read data in FASTQ format.</p>
<p>You will be working with an Illumina MiSeq read dataset from a genome sequence project. The sequenced organism is an enterohaemorrhagic E. coli (EHEC) of the serotype O157, a potentially fatal gastrointestinal pathogen. The sequenced bacterium was part of an outbreak investigation in the St. Louis area, USA in 2011.
The sequencing was done as paired-end 2x150bp.</p>
<h2 id="downloading-the-data">Downloading the data</h2>
<p>The raw data were deposited at the European Nucleotide Archive, under the accession number SRR957824. Go to the ENA <a href="http://www.ebi.ac.uk/ena">website</a> and search for the run with the accession SRR957824. Download the two fastq files associated with the run:</p>
<pre><code>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR957/SRR957824/SRR957824_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR957/SRR957824/SRR957824_2.fastq.gz
</code></pre>

<h2 id="fastqc">FastQC</h2>
<p>To check the quality of the sequence data we will use a tool called FastQC. With this you can check things like read length distribution, quality distribution across the read length, sequencing artifacts and much more.</p>
<p>FastQC has a graphical interface and can be downloaded and run on a Windows or Linux computer without installation. It is available <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here</a>.</p>
<p>However, FastQC is also available as a command line utility on the training server you are using. You can load the module and execute the program as follows:</p>
<pre><code>module load FastQC
fastqc $read1 $read2
</code></pre>

<p>which will produce both a .zip archive containing all the plots, and a html document for you to look at the result in your browser.</p>
<p>Open the html file with your favourite web browser, and try to interpret them. </p>
<p>Pay special attention to the per base sequence quality and sequence length distribution. Explanations for the various quality modules can be found <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">here</a>. Also, have a look at examples of a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html">good</a> and a <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html">bad</a> illumina read set for comparison.</p>
<p>You will note that the reads in your uploaded dataset have fairly poor quality (&lt;20) towards the end. There are also outlier reads that have very poor quality for most of the second half of the reads.</p>
<p>There are overrepresented sequences in the data. Where do they come from?</p>
<h2 id="scythe">Scythe</h2>
<p>Scythe uses a Naive Bayesian approach to classify contaminant substrings in sequence reads. It considers quality information, which can make it robust in picking out 3'-end adapters, which often include poor quality bases.</p>
<p>First, install scythe:</p>
<pre><code>git clone https://github.com/vsbuffalo/scythe.git
cd scythe
make all
</code></pre>

<p>Then, copy or move "scythe" to a directory in your $PATH, for example like this:</p>
<p><code>cp scythe $HOME/bin/</code></p>
<p>Scythe can be run minimally with:</p>
<p><code>scythe -a adapter_file.fasta -o trimmed_sequences.fastq sequences.fastq</code></p>
<p>Try to trim the adapters in both your read files!</p>
<h2 id="sickle">Sickle</h2>
<p>Most modern sequencing technologies produce reads that have deteriorating quality towards the 3'-end and some towards the 5'-end as well. Incorrectly called bases in both regions negatively impact assembles, mapping, and downstream bioinformatics analyses.</p>
<p>We will trim each read individually down to the good quality part to keep the bad part from interfering with downstream applications.</p>
<p>To do so, we will use sickle. Sickle is a tool that uses sliding windows along with quality and length thresholds to determine when quality is sufficiently low to trim the 3'-end of reads and also determines when the quality is sufficiently high enough to trim the 5'-end of reads. It will also discard reads based upon a length threshold.</p>
<p>First, install sickle:</p>
<pre><code>git clone https://github.com/najoshi/sickle.git
cd sickle
make
</code></pre>

<p>Copy sickle to a directory in your $PATH:</p>
<p><code>cp sickle $HOME/bin/</code></p>
<p>Sickle has two modes to work with both paired-end and single-end reads: sickle se and sickle pe.</p>
<p>Running sickle by itself will print the help:</p>
<p><code>sickle</code></p>
<p>Running sickle with either the "se" or "pe" commands will give help specific to those commands. Since we have paired end reads:</p>
<p><code>sickle pe</code></p>
<p>Set the quality score to 25. This means the trimmer will work its way from both ends of each read, cutting away any bases with a quality score &lt; 25.</p>
<pre><code>sickle pe -f input_file1.fastq -r input_file2.fastq -t sanger \
-o trimmed_output_file1.fastq -p trimmed_output_file2.fastq \
-s trimmed_singles_file.fastq -q 25
</code></pre>

<p>What did the trimming do to the per-base sequence quality, the per sequence quality scores and the sequence length distribution? Run FastQC again to find out.</p>
<p>What is the sequence duplication levels graph about? Why should you care about a high level of duplication, and why is the level of duplication very low for this data?</p>
<p>Based on the FastQC report, there seems to be a population of shorter reads that are technical artifacts. We will ignore them for now as they will not interfere with our analysis.</p>
<h2 id="extra-exercises">Extra exercises</h2>
<p>Perform quality control on the extra datasets given by your instructors. </p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../pan_genome/" title="Pan-genome analysis" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Pan-genome analysis
              </span>
            </div>
          </a>
        
        
          <a href="../rna_seq/" title="Rna seq" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Rna seq
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../assets/javascripts/application-8e4952e681.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:".."}})</script>
      
    
    
      
    
  </body>
</html>